flexmeasures/api/v3_0/sensors.py

```diff
@@ -154,3 +189,347 @@ def get_data(self, response: dict):
+                    }
+                ],
+                "soc_min": 10,
+                "soc_max": 25,
+                "roundtrip_efficiency": 0.98
+            }
+
+        **Example response**
+
+        This message indicates that the scheduling request has been processed without any error.
+        A scheduling job has been created with some Universally Unique Identifier (UUID),
+        which will be picked up by a worker.
+        The given UUID may be used to obtain the resulting schedule: see /sensors/<id>/schedules/<uuid>.
+
+        .. sourcecode:: json
+
+            {
+                "status": "PROCESSED",
+                "schedule": "364bfd06-c1fa-430b-8d25-8f5a547651fb",
+                "message": "Request has been processed."
+            }
+
+        :reqheader Authorization: The authentication token
+        :reqheader Content-Type: application/json
+        :resheader Content-Type: application/json
+        :status 200: PROCESSED
+        :status 400: INVALID_TIMEZONE, INVALID_DATETIME, INVALID_DOMAIN, INVALID_UNIT, PTUS_INCOMPLETE
+        :status 401: UNAUTHORIZED
+        :status 403: INVALID_SENDER
+        :status 405: INVALID_METHOD
+        """
+        # todo: if a soc-sensor entity address is passed, persist those values to the corresponding sensor
+        #       (also update the note in posting_data.rst about flexibility states not being persisted).
+
+        # get value
+        if "value" not in kwargs:
+            return ptus_incomplete()
+        try:
+            value = float(kwargs.get("value"))  # type: ignore
+        except ValueError:
+            extra_info = "Request includes empty or ill-formatted value(s)."
+            current_app.logger.warning(extra_info)
+            return ptus_incomplete(extra_info)
+        if unit == "kWh":
+            value = value / 1000.0
+
+        # Convert round-trip efficiency to dimensionless
+        if roundtrip_efficiency is not None:
+            roundtrip_efficiency = roundtrip_efficiency.to(
+                ur.Quantity("dimensionless")
+            ).magnitude
+
+        # get optional min and max SOC
+        soc_min = kwargs.get("soc_min", None)
+        soc_max = kwargs.get("soc_max", None)
+        if soc_min is not None and unit == "kWh":
+            soc_min = soc_min / 1000.0
+        if soc_max is not None and unit == "kWh":
+            soc_max = soc_max / 1000.0
+
+        # set soc targets
+        end_of_schedule = start_of_schedule + current_app.config.get(  # type: ignore
+            "FLEXMEASURES_PLANNING_HORIZON"
+        )
+        resolution = sensor.event_resolution
+        soc_targets = pd.Series(
+            np.nan,
+            index=pd.date_range(
+                start_of_schedule, end_of_schedule, freq=resolution, closed="right"
+            ),  # note that target values are indexed by their due date (i.e. closed="right")
+        )
+        for target in kwargs.get("targets", []):
```

Comment:
I guess factoring out the parsing of target value and datetime into the `TargetSchema` would be the refactoring which would make this endpoint much more readable.